utils.pwsh/Setup-Host.ps1

```diff
@@ -0,0 +1,45 @@
+function Setup-Host {
+    if ( ! ( Test-Path function:Log-Output ) ) {
+        . $PSScriptRoot/Logger.ps1
+    }
+
+    if ( ! ( Test-Path function:Check-Git ) ) {
+        . $PSScriptRoot/Check-Git.ps1
+    }
+
+    Check-Git
+
+    if ( ! ( Test-Path function:Invoke-External ) ) {
+        . $PSScriptRoot/Invoke-External.ps1
+    }
+
+    try {
+        $script:ProjectRoot = Invoke-External git rev-parse --show-toplevel 2>$null
+    } catch {
+        Log-Warning "Not running in a git repository, interpreting project root instead"
+        $script:ProjectRoot = $($script:PSScriptRoot)
+    }
+
+    $script:WorkRoot = "${ProjectRoot}\windows_build_temp"
```

Comment:
I notice that this uses `windows_build_temp` rather than the current `windows_native_build_temp`. Was this intentional?

Reply:
Yes, didn't see a reason to call it specifically the "native" build temp. Every download/dependency is access specifically in that directory, so sharing it between Windows and WSL (if necessary) shouldn't be an issue.

Reply:
I think I mostly did this originally to avoid clobbering any of the cross-compiled stuff, particularly mbedtls, which I knew we had to build twice, and nasm, which we built manually to set up the cross-compile environment but then also deployed a prebuilt version on Windows (which I will be correcting after this lands).

If you think this is safe enough to do, I won't worry about it here. If we find issues with it later, we can address it later.